qt/python/mantidqt/widgets/superplot/test/test_superplot_presenter.py

```diff
@@ -0,0 +1,308 @@
+        calls = [mock.call("ws1", 10), mock.call("ws2", 10)]
+        self.m_model.add_data.assert_has_calls(calls)
+        self.m_model.set_spectrum_mode.assert_called_once()
+        self.presenter._update_list.assert_called_once()
+        self.presenter._update_plot.assert_called_once()
+        self.m_view.set_selection.assert_called_once_with({"ws1": [1],
+                                                           "ws2": [2]})
+        self.m_view.get_mode.return_value = self.presenter.BIN_MODE_TEXT
+        self.presenter._on_hold()
+        self.m_model.set_bin_mode.assert_called_once()
+
+    def test_on_un_hold(self):
+        self.presenter._update_list = mock.Mock()
+        self.presenter._update_plot = mock.Mock()
+        self.presenter._update_spectrum_slider = mock.Mock()
+        self.m_view.is_spectrum_disabled.return_value = False
+        self.m_view.get_selection.return_value = {"ws1": [], "ws2": []}
+        self.m_view.get_spectrum_slider_position.return_value = 10
+        self.presenter._on_un_hold()
+        calls = [mock.call("ws1", 10), mock.call("ws2", 10)]
+        self.m_model.remove_data.assert_has_calls(calls)
+        self.presenter._update_list.assert_called_once()
+        self.presenter._update_plot.assert_called_once()
+        self.presenter._update_spectrum_slider.assert_called_once()
+
+    def test_on_hold_button_toggled(self):
+        self.presenter._on_hold = mock.Mock()
+        self.presenter._on_un_hold = mock.Mock()
+        self.presenter.on_hold_button_toggled(True)
+        self.presenter._on_hold.assert_called_once()
+        self.presenter._on_un_hold.assert_not_called
+        self.presenter.on_hold_button_toggled(False)
+        self.presenter._on_un_hold.assert_called_once()
+
+    def test_on_mode_changed(self):
+        self.presenter._update_spectrum_slider = mock.Mock()
+        self.presenter._update_plot = mock.Mock()
+        self.m_view.get_selection.return_value = {"ws1": [1], "ws2": [2]}
+        self.presenter.on_mode_changed("mode")
+        self.presenter._update_spectrum_slider.assert_called_once()
+        self.presenter._update_plot.assert_called_once()
+
+    def test_on_workspace_releted(self):
+        self.presenter._update_list = mock.Mock()
+        self.presenter._update_plot = mock.Mock()
+        self.m_view.get_selection.return_value = {"ws1": [1], "ws2": [2]}
+        self.presenter.on_workspace_deleted("ws1")
+        self.m_view.set_selection.assert_called_once_with({"ws2": [2]})
+        self.presenter._update_list.assert_called_once()
+        self.presenter._update_plot.assert_called_once()
+
+    def test_on_workspace_renamed(self):
+        self.presenter._update_list = mock.Mock()
+        self.presenter._update_plot = mock.Mock()
+        self.m_view.get_selection.return_value = {"ws1": [1], "ws2": [2]}
+        self.presenter.on_workspace_renamed("ws1", "ws3")
+        self.m_view.set_selection.assert_called_once_with({"ws3": [1],
```

Comment:
The `m_` in `self` variables is not necessary here.

Reply:
`m_` means mock, not member.